### DISCUSSION TITLE
Two belongsToMany relationships in one?

### DISCUSSION THREAD

**Discussion by joshstobbs** [staudenmeir/eloquent-has-many-deep#230] [Q&A] [answered]
I have a scenario in which I am trying to create a merged relationship out of two belongsToMany relationships - it looks something like this:

User model:

```
public function followedProjects(): BelongsToMany
{
    return $this->belongsToMany(Project::class, 'project_follower', 'follower_id', 'project_id');
}

public function assignedProjects(): BelongsToMany
{
    return $this->belongsToMany(Project::class, 'project_designer', 'designer_id', 'project_id');
}
```

Ideally I'd like to be able to call something like `$user->projects` and retrieve results from both pivots.  Is that possible with this package?

**Comment by staudenmeir** [comment, marked as answer]
Hi @joshstobbs,
You can merge these relationships with another package of mine:
https://github.com/staudenmeir/laravel-merged-relations

```php
use Staudenmeir\LaravelMergedRelations\Facades\Schema;

Schema::createMergeView(
    'all_projects',
    [(new User)->followedProjects(), (new User)->assignedProjects()]
);
```

```php
class User extends Model
{
    use \Staudenmeir\LaravelMergedRelations\Eloquent\HasMergedRelationships;

    public function projects()
    {
        return $this->mergedRelationWithModel(Project::class, 'all_projects');
    }
}
```

**Comment by erikjandelange** [comment]
Hi @staudenmeir 

Edit: 
Don't if you prefer i create an issue for it in the other repository, but this was a simular situation. 
If so, i'll create it, no prob.

This works great but `withPivot()` over **two different pivottables** the view is not correct.

Using the example above:

```
public function followedProjects(): BelongsToMany
{
    return $this->belongsToMany(Project::class, 'project_follower', 'follower_id', 'project_id')
        ->withPivot('sorting');
}

public function assignedProjects(): BelongsToMany
{
    return $this->belongsToMany(Project::class, 'project_designer', 'designer_id', 'project_id')
        ->withPivot('sorting');
}
```

To create an example of the views:

```
Schema::createMergeView(
    'followers',
    [
        (new User)->followedProjects()->withPivot('sorting'),
    ]
);
Schema::createMergeView(
    'designers',
    [
        (new User)->assignedProjects()->withPivot('sorting')
    ]
);
Schema::createMergeView(
    'all',
    [
        (new User)->followedProjects()->withPivot('sorting'),
        (new User)->assignedProjects()->withPivot('sorting')
    ]
);
```

The `all` version doesnt have the pivot field while the other two have different pivotfieldnames.

`__project_follower__pivot__sorting` and `__project_designer__pivot__sorting`

How to solve this issue?

**Reply by staudenmeir** [reply]
@erikjandelange Different pivot tables are not supported yet. Please create an issue in the other repository.

**Comment by erikjandelange** [comment]
Thanks for the quick answer.
I've [created an issue](https://github.com/staudenmeir/laravel-merged-relations/issues/20) as suggested.
